dali/imgcodec/decoders/nvjpeg/nvjpeg_test.cc

```diff
@@ -72,6 +72,32 @@ class NvJpegDecoderTest : public NumpyDecoderTestBase<GPUBackend, OutputType> {
     return std::make_shared<JpegParser>();
   }
 
+  /**
+   * @brief Checks if the image is similar to the reference.
+   *
+   * Checks if 1) max error is lower than 30% 2) mean squared error is lower than 1%.
+   */
+  void AssertSimilar(const TensorView<StorageCPU, const OutputType> &img,
+                     const Tensor<CPUBackend> &ref_tensor) {
+    TYPE_SWITCH(ref_tensor.type(), type2id, RefType, NUMPY_ALLOWED_TYPES, (
+      auto ref = view<const RefType>(ref_tensor);
+
+      float eps = ConvertSatNorm<OutputType>(0.3);
+      Check(img, ref, EqualConvertNorm(eps));
+
+      double mean_error = 0;
+      uint64_t size = volume(img.shape);
+      for (size_t i = 0; i < size; i++) {
+        double img_value = ConvertSatNorm<double>(img.data[i]);
+        double ref_value = ConvertSatNorm<double>(ref.data[i]);
+        double error = abs(img_value - ref_value);
+        mean_error += error;
```

Comment:
Mean square, as mentioned?
```suggestion
        double error = img_value - ref_value;
        mean_error += error * error;
```